server/athenian/api/align/mutations/goals.py

```diff
@@ -259,11 +288,65 @@ def _ids_repr(ids: Sequence[int]) -> str:
                 await parse_request_repositories(value, request, account_id), False,
             )
 
+    strings_field_parser = _StringsListUpdateInfoParser(input)
+    jira_projects = strings_field_parser(UpdateGoalInputFields.jiraProjects, lambda v: v)
+    jira_priorities = strings_field_parser(
+        UpdateGoalInputFields.jiraPriorities, _parse_request_jira_priorities_value,
+    )
+    jira_issue_types = strings_field_parser(
+        UpdateGoalInputFields.jiraIssueTypes, _parse_request_jira_issue_types_value,
+    )
+
     return GoalUpdateInfo(
         team_goal_deletions=deletions,
         team_goal_assignments=assignments,
         archived=input.get(UpdateGoalInputFields.archived),
         name=input.get(UpdateGoalInputFields.name),
         metric=input.get(UpdateGoalInputFields.metric),
         repositories=repositories,
+        jira_projects=jira_projects,
+        jira_priorities=jira_priorities,
+        jira_issue_types=jira_issue_types,
     )
+
+
+class _StringsListUpdateInfoParser:
+    def __init__(self, input: dict[str, Any]):
+        self._input = input
+
+    def __call__(
+        self,
+        field_name: str,
+        _parse_value: Callable[[list[str]], list[str]],
+    ) -> Optional[StringsListUpdateInfo]:
+        if (field := self._input.get(field_name)) is None:
+            return None
+        if (value := field.get(UpdateRepositoriesInputFields.value)) is None:
+            return StringsListUpdateInfo([], True)
+        return StringsListUpdateInfo(_parse_value(value), False)
+
+
+def _parse_request_jira_projects(input: dict[str, Any]) -> Optional[list[str]]:
+    return input.get(CreateGoalInputFields.jiraProjects)
+
+
+def _parse_request_jira_priorities(input: dict[str, Any]) -> Optional[list[str]]:
+    if (priorities := input.get(CreateGoalInputFields.jiraPriorities)) is None:
+        return priorities
+    else:
+        return _parse_request_jira_priorities_value(priorities)
+
+
+def _parse_request_jira_priorities_value(value: list[str]) -> list[str]:
+    return list(dict.fromkeys([normalize_priority(p) for p in value]))
```

Comment:
```suggestion
    return list({normalize_priority(p) for p in value})
```

Reply:
`fromkeys` was used to preserve while dropping duplicates, i.e. if the user selects [Tasks, Bugs] it is not good if he sees [Bugs,Tasks] after server round-trip

Reply:
We should sort the values alphabetically before returning them to the user, good observation. As for the original order, it's not important here.

Reply:
changed to `sorted(set(...` on when writing